### PR TITLE
Mobile updates

### PR DESCRIPTION
**Pull request by nevgauker**
# Description

Some  simple layout changes so  both dashboard charts and nav bar will be visible on mobile
Dismiss the side menu on mobile when clicking

#16 

Screenshot 2023-10-12 at 17 04 17

Screenshot 2023-10-12 at 17 04 26

* still has some issues with the historical schedule results.

* I think the side menu should be a popup and not part of the page layout

### CODE REVIEW

File: src/pages/components/header/Header.tsx [hunk 1]
```diff
@@ -5,6 +5,9 @@ import cn from 'classnames'
 export default function Header({ pages }: { pages: MenuPage[] }) {
   const [open, setOpen] = useState(false)
 
+  function dismiss() {
+    setOpen(false)
+  }
```
Comment (NoamGaash) [inline src/pages/components/header/Header.tsx:10]:
```suggestion
  useCallback(function dismiss() {
    setOpen(false)
  }, [])
```

File: src/pages/components/header/menu/Menu.tsx [hunk 2]
```diff
@@ -3,25 +3,46 @@ import { useLocation, useNavigate } from 'react-router-dom'
...
   const navigate = useNavigate()
   const { pathname: currpage } = useLocation()
+  const [isMobile, setIsMobile] = useState(window.innerWidth <= 768)
+
   console.log({ currpage, active: pages.map((p) => p.key === currpage) })
 
+  function navigateTo(page: MenuPage, dismiss: boolean) {
+    page.key[0] === '/' ? navigate(page.key) : window.open(page.key, '_blank')
+    if (dismiss) {
+      shouldDismiss()
```
Comment (NoamGaash) [inline src/pages/components/header/menu/Menu.tsx:23]: I think the naming convention is confusing
shouldDismiss is the funtion that dismiss the menu, and dismiss is the boolean that tell us whether the menu should be dismissed?

File: src/pages/components/header/menu/Menu.tsx [hunk 3]
```diff
@@ -3,25 +3,46 @@ import { useLocation, useNavigate } from 'react-router-dom'
...
 export type MenuPage = {
   label: string
   key: string
 }
 
-function Menu({ pages }: { pages: MenuPage[] }) {
+function Menu({ pages, shouldDismiss }: { pages: MenuPage[]; shouldDismiss: () => void }) {
   const navigate = useNavigate()
   const { pathname: currpage } = useLocation()
+  const [isMobile, setIsMobile] = useState(window.innerWidth <= 768)
```
Comment (NoamGaash) [inline src/pages/components/header/menu/Menu.tsx:16]:
```suggestion
  const isMobile= window.innerWidth <= 768
```
state is not needed

File: yarn.lock [hunk 4]
```diff
@@ -4221,10 +4221,10 @@ typed-array-length@^1.0.4:
...
     is-typed-array "^1.1.9"
 
-typescript@^4.7.4:
-  version "4.9.5"
-  resolved "https://registry.yarnpkg.com/typescript/-/typescript-4.9.5.tgz#095979f9bcc0d09da324d58d03ce8f8374cbe65a"
-  integrity sha512-1FXk9E2Hm+QzZQ7z+McJiHL4NW1F2EzMu9Nq9i3zAaGqibafqYwCVU6WyWAuyQRRzOlxou8xZSyXLEN8oKj24g==
+typescript@^5.2.2:
+  version "5.2.2"
+  resolved "https://registry.yarnpkg.com/typescript/-/typescript-5.2.2.tgz#5ebb5e5a5b75f085f22bc3f8460fba308310fa78"
+  integrity sha512-mI4WrpHsbCIcwT9cF4FZvr80QUeKvsUsUvKDoR+X/7XHQH98xYD8YHZg7ANtz2GtZt/CBq2QJ0thkGJMHfqc1w==
```
Comment (NoamGaash) [inline yarn.lock:4227]: should be on seperate PR

**Review by NoamGaash — CHANGES_REQUESTED** [review]
I'm not exactly sure what behaviour you're trying to achieve, but let's place the opening logic inside the header or inside the menu, but I see no reason to place it in both of the components

File: src/pages/components/header/menu/Menu.tsx (hunk 2 above)
Comment (nevgauker) [reply src/pages/components/header/menu/Menu.tsx:23]: I agree, no idea what I was thinking :) the name will be switched in order to make sense

File: yarn.lock (hunk 4 above)
Comment (nevgauker) [reply yarn.lock:4227]: No idea what changed. no new package was installed

**Comment by NoamGaash** [comment]
@nevgauker hi, how things are going? do you need any help?

**Comment by NoamGaash** [comment]
please re-open if you're still on it

**Pull request closed without merging**